app/javascript/mastodon/components/intersection_observer_article.js

```diff
@@ -1,10 +1,15 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import ImmutablePureComponent from 'react-immutable-pure-component';
 import scheduleIdleTask from '../features/ui/util/schedule_idle_task';
 import getRectFromEntry from '../features/ui/util/get_rect_from_entry';
+import { is } from 'immutable';
 
-export default class IntersectionObserverArticle extends ImmutablePureComponent {
+// Diff these props in the "rendered" state
+const updateOnPropsForRendered = ['id', 'index', 'listLength'];
+// Diff these props in the "unrendered" state
+const updateOnPropsForUnrendered = ['id', 'index', 'listLength', 'cachedHeight'];
```

Comment:
`children` seems to never change. `saveHeightKey` doesn't matter because it's not rendered; it's just used in the `handleIntersection` function. `onHeightChange` is a function. `intersectionObserverWrapper` never changes. I believe the props listed above are the only ones that matter for diffing.